provider_test.go

```diff
@@ -3,12 +3,28 @@ package main
 import (
 	"errors"
 	"testing"
+	"strings"
 
 	"github.com/coreos/fleet/client"
-
 	"github.com/hashicorp/terraform/helper/schema"
 )
 
+// retry wraps a function with retry logic. Only errors containing "timed out"
+// will be retried. (authentication errors and other stuff should fail
+// immediately)
+func retry(f func() (interface{}, error), maxRetries int) (interface{}, error) {
+	var result interface{}
+	var err error
+
+	for retries := 0; retries < maxRetries; retries++ {
+		result, err = f()
+		if err == nil || !strings.Contains(err.Error(), "timed out") {
+			break
+		}
+	}
+	return result, err
+}
+
```

Comment:
Does this belong in the tests? It originally existed to support a use case where Terraform provisioned the EC2 instance, but we had to wait for it to start up before we could proceed with creating Fleet units via an SSH tunnel connection. Looks like that functionality no longer exists, but given that the original case no longer works anyway, that may not be a bad thing...